witchcraft/internal/tcpjson/tcp_logger.go

```diff
@@ -88,6 +95,14 @@ func (d *TCPWriter) Write(p []byte) (int, error) {
 	return len(p), nil
 }
 
+// trimNewLine will remove a trailing new line character and return a new byte slice.
+func trimNewLine(b []byte) []byte {
```

Comment:
The https://golang.org/pkg/bytes/#TrimSuffix function basically does this -- can do `bytes.TrimSuffix(b, []byte("\n"))` in place of calling this function.

Reply:
I intentionally wrote this to avoid allocating yet another `[]byte`, but in the grand scheme of things feels like an over optimization. Went ahead and changed it to the stdlib function.

Reply:
By allocation, do you mean the `[]byte("\n")` line (since I don't believe the implementation of the function performs any allocations)? If we think that's an issue, then fine with going with this function, although would explain that in a comment. If the allocation is the only concern we could also have a package-level `var newlineBytes = []byte("\n")` that is allocated once and used in the trim operation

Reply:
Yep I was referring to the `[]byte("\n")`. I ran a quick benchmark to check my hypothesis and it turns out inlining the `[]byte("\n")` does not actually allocate, so no need for the local var.